src/App.tsx

```diff
@@ -1,39 +1,67 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = this.setTimer();
+    document.addEventListener('click', this.handleClick);
+    document.addEventListener('contextmenu', this.handleContextmenu);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+  componentDidUpdate(_: Readonly<{}>, prevState: Readonly<State>) {
+    const prevName = prevState.clockName;
+    const newName = this.state.clockName;
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    if (prevName !== newName) {
+      // eslint-disable-next-line
+      console.log(`Renamed from ${prevName} to ${newName}`);
+    }
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  setTimer = () => {
+    return window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  };
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  handleClick = () => {
+    this.timerId = this.setTimer();
+    this.setState({ hasClock: true });
+  };
 
-        {' time is '}
+  handleContextmenu = () => {
+    this.setState({ hasClock: false });
+    window.clearInterval(this.timerId);
```

Comment:
It's overcomplicated. You don't need to clear and create new timer for calling `getRandomName`

Reply:
Methods `handleContextmenu` and `handleClick` should change only visibility of the clock